src/main/java/hudson/plugins/promoted_builds/integrations/jobdsl/ConditionsContext.java

```diff
@@ -0,0 +1,171 @@
+package hudson.plugins.promoted_builds.integrations.jobdsl;
+
+import groovy.lang.Closure;
+import groovy.util.Node;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javaposse.jobdsl.dsl.Context;
+import javaposse.jobdsl.dsl.FileJobManagement;
+import javaposse.jobdsl.dsl.helpers.BuildParametersContext;
+
+import org.apache.commons.io.FileUtils;
+
+public class ConditionsContext implements Context {
+
+	// Self Promotion Condition
+	private boolean isSelfPromotion = false;
+	
+	private boolean evenIfUnstable = false;
+	
+	// Parametzerized Self Promotion Condition
+	private boolean isParameterizedSelfPromotion = false;
+	
+	private boolean evenIfUnstableParameterized = false;
+	
+	private String parameterName = null;
+	
+	private String parameterValue = null;
+	
+	// Manual Promotion Condition
+	private boolean isManual = false;
+	
+	private String users = null;
+	
+	final Map<String, Node> params = new HashMap<String, Node>();
+	
+	// Release Build Condition
+	private boolean isReleaseBuild = false;
+	
+	// Downstream Build Condition
+	private boolean isDownstreamPass = false;
+	
+	private boolean evenIfUnstableDownstream = false;
+	
+	private String jobs = null;
+	
+	// Upstream Build Condition
+	private boolean isUpstreamPromotion = false;
+	
+	private String promotionNames = null;
+
+	public void selfPromotion(Boolean evenIfUnstable) {
+		isSelfPromotion = true;
+		if (evenIfUnstable) {
+			this.evenIfUnstable = evenIfUnstable;
+		}
+	}
+
+	public void parameterizedSelfPromotion(Boolean evenIfUnstable, String parameterName, String parameterValue) {
+		isParameterizedSelfPromotion = true;
+		if (evenIfUnstable) {
+			this.isParameterizedSelfPromotion = evenIfUnstable;
+		}
+		this.parameterName = parameterName;
+		this.parameterValue = parameterValue;
+	}
+
+	public void manual(String users) {
+		isManual = true;
+		this.users = users;
+	}
+
+	public void manual(String users, Closure<?> parametersClosure) {
+		isManual = true;
+		this.users = users;
+		parameters(parametersClosure);
+	}
+
+	public void releaseBuild() {
+		isReleaseBuild = true;
+	}
+
+	public void parameters(Closure<?> parametersClosure) {
+		// delegate to main BuildParametersContext
+		BuildParametersContext parametersContext = new BuildParametersContext(
```

Comment:
This is a but quirky. Using `FileJobManagment` here is a workaround for missing access to `JobManagement`. I will add a method to `DslEnvironment` so that an extension can instantiate contexts. See https://github.com/jenkinsci/job-dsl-plugin/pull/772. This will be in job-dsl 1.44 which I hope to release soon.